crates/edr_provider/src/data.rs

```diff
@@ -433,29 +488,38 @@ impl ProviderData {
                 .expect("If the transaction was inserted in a block, it must have a receipt")
                 .transaction_index;
 
-            let tx_index_usize =
+            let tx_index =
                 usize::try_from(tx_index).expect("Indices cannot be larger than usize::MAX");
-            let signed_transaction = tx_block.transactions()[tx_index_usize].clone();
-
-            let block_metadata = BlockMetadataForTransaction {
-                base_fee_per_gas: tx_block.header().base_fee_per_gas,
-                block_hash: *tx_block.hash(),
-                block_number: tx_block.header().number,
-                transaction_index: tx_index,
-            };
 
-            Some(GetTransactionResult {
-                signed_transaction,
-                spec_id: self.blockchain.spec_id(),
-                block_metadata: Some(block_metadata),
-            })
+            self.transaction_from_block(&tx_block, tx_index)
         } else {
             None
         };
 
         Ok(transaction)
     }
 
+    pub fn transaction_from_block(
```

Comment:
I would have expected this logic to happen in the rpc handler, not here.

Reply:
My thinking with the `GetTransactionResult` (for which `transaction_from_block` is a helper method) is this: 

The transaction getter RPC methods need some data from the block (header stuff and the transaction index) in addition to the signed transaction. To avoid cloning entire blocks just to get a transaction, the `GetTransactionResult` type exposes just the data needed by the transaction getter RPC methods in the internal representation (e.g. block number as `u64`) and then lets the RPC methods convert to the data types expected in the RPC layer.

Reply:
Actually on second thought we cloning the block is cheap, because it's behind an `Arc`. But we still need a structure to return both the signed transaction and the block and I prefer explicit types.